src/App.js

```diff
@@ -1,26 +1,58 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, {Component} from 'react';
+import classes from './App.module.scss';
+import GameCardList from './Components/GameCardList/GameCardList';
+import { BrowserRouter } from 'react-router-dom';
+import axios from './axios';
+import Spinner from './Components/Spinner/Spinner';
+
+class App extends Component {
+  state = {
+    games: [],
+    categoryOptions: [],
+    loading: false
+  }
+
+  componentDidMount = async () => {
+    const categoryOptions = await axios.get('categories')
+    this.setState({categoryOptions: categoryOptions.data.categories})
+  }
+
+  categoryChangeHandler = async (event) => {
+    event.preventDefault();
+    this.setState({loading: true})
+    const categoryGameData = await axios.get(`search?category=${event.target.value}`)
+    this.setState({loading: false, games: await categoryGameData.data.games })
+  }
+
+
+  render() {
+    let categories = <option>Loading...</option>;
+    if(this.state.categoryOptions.length > 0) {
+      categories = this.state.categoryOptions.map(category => {
+        return <option value={category.id} key={category.id}>
+          {category.name}
+        </option>
+      })
+    }
+
+    let gameCards = null;
+    if(this.state.loading === true) {
```

Comment:
This could be just `this.state.loading` as it will either be true or false maybe?